ports/rp2/mpconfigport.h

```diff
@@ -86,12 +86,14 @@
 #endif
 
 // Memory allocation policies
+#ifndef MICROPY_GC_STACK_ENTRY_TYPE
```

Comment:
Is there a reason for this change in particular? Defining `MICROPY_GC_STACK_ENTRY_TYPE` would disable PSRAM unconditionally, which seems counter-intuitive.

Possibly should be moved to the `#else` branch of `#if MICROPY_HW_ENABLE_PSRAM`?

In cases where someone configures their board with:

```
#define MICROPY_HW_ENABLE_PSRAM                   1
#define MICROPY_GC_STACK_ENTRY_TYPE             uint16_t
```

Then we should probably fail the build with an error. An example of that in practise is:

https://github.com/micropython/micropython/blob/398da22492d9e593f87f0b9ae4fc043f8b2b8a11/ports/rp2/rp2_psram.h#L34-L37